apisix/admin/init.lua

```diff
@@ -117,10 +117,13 @@ local function run()
         core.response.exit(404)
     end
 
-    ngx.req.read_body()
-    local req_body = ngx.req.get_body_data()
-
+    local req_body = core.request.get_body()
     if req_body then
+        if #req_body > 1024 * 1024 * 1.5 then
```

Comment:
We need to detect the size (use `stat`, for example) before reading it. It doesn't make sense to do the check if we have already read the very BIG file.

Reply:
agreed.